src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -283,7 +278,7 @@ CreateSingleTaskRouterPlan(DistributedPlan *distributedPlan, Query *originalQuer
 	distributedPlan->workerJob = job;
 	distributedPlan->masterQuery = NULL;
 	distributedPlan->routerExecutable = true;
-	distributedPlan->hasReturning = false;
+	distributedPlan->expectResults = query->commandType == CMD_SELECT;
```

Comment:
1. Is it guaranteed that here `query->returningList == NIL` when `query->cmdType != CMD_SELECT`?
2. How come returning always `false` didn't cause problems before, even for SELECT and UPDATE ... RETURNING?

Reply:
1. not sure, it's likely a bug, will try see if I can hit it, will make a separate PR if I find anything since the goal of this PR is to change nothing
2. previously had a `|| modLevel == MODIFY_READONLY` check included when later computing `expectResults`

Reply:
Looked into 1. `CreateDistributedPlan` only calls this function when not a modifying command. So maybe this should be an `Assert(query->commandType == CMD_SELECT)` instead